### DISCUSSION TITLE
laser (OSTech diode) remains enabled

### DISCUSSION THREAD

**Discussion by wrguenthner** [NMGRL/pychron#2184] [Q&A]
Periodically the laser in Pydiode will remain "enabled" after Pychron has finished an extraction. This then creates subsequent issues when Pychron tries to initiate another laser extraction, typically manifested as a failure to connect. Trying to disable the laser directly in Pydiode leads to the program freezing. I have not observed this behavior enough to see a consistent sequence of events that creates the issue. For reference, the most recent occurrence in our logs was at 10:26 am on 11/18/22.

**Comment by jirhiker** [comment]
All elements of that sound like pydiode loss communications with the laser.  

can you post a log file that displays this behavior  here

**Reply by wrguenthner** [reply]
Attached is a relevant log that shows the behavior, which can be found at the 2022-11-15 12:05:53,566 timestamp. Yes, it looks to be a timed out error on the ethernet communicator. Is this a pychron issue or is it related to our network (perhaps a firewall issue)?
[10000-2623-001.log](https://github.com/NMGRL/pychron/files/10115424/10000-2623-001.log)

**Reply by jirhiker** [reply]
I don't remember exactly but it seems like you are running two instances of pychron, one for valve control etc, and one for laser control?   can you send the log from the laser computer?

**Reply by wrguenthner** [reply]
sorry for the slow reply on this... here is the relevant log for this from the laser computer. Yes, that's correct. We're running the laser with one computer and the other for valve control, mass spec, etc.
[pychron-00057.log](https://github.com/NMGRL/pychron/files/10260756/pychron-00057.log)
